File: SMART-AGRICULTURE-IOT-master/src/models/location.model.ts
import mongoose, { Document, Schema } from 'mongoose';

export interface ILocation extends Document {
  name: string;
  location_code: string; 
  description: string;
  area: string;
  seasonId: mongoose.Types.ObjectId;
  sensorDataId: mongoose.Types.ObjectId;
  sensor_settingsId: mongoose.Types.ObjectId;
  created_at: Date;
  updated_at: Date;
}

const LocationSchema: Schema = new Schema({
  name: { type: String, required: true },
  location_code: { type: String, required: true, unique: true }, 
  description: { type: String },
  area: { type: String },
  seasonId: { type: Schema.Types.ObjectId, ref: 'Season', required: true },
  sensorDataId: { type: Schema.Types.ObjectId, ref: 'SensorData' },
  sensor_settingsId: { type: Schema.Types.ObjectId, ref: 'SensorSetting' },
  created_at: { type: Date, default: Date.now },
  updated_at: { type: Date, default: Date.now }
});

LocationSchema.index({ location_code: 1 }, { unique: true });

export default mongoose.model<ILocation>('Location', LocationSchema);

File: lib/models/location_model.dart
class LocationModel {
  final String id;
  final String name;
  final String description;
  final String area;
  final String seasonId;
  LocationModel({
    required this.id,
    required this.name,
    required this.description,
    required this.area,
    this.seasonId = "",
  });
  factory LocationModel.fromJson(Map<String, dynamic> json) {
    return LocationModel(
      id: json['_id'] ?? "",
      name: json['name'] ?? "",
      description: json['description'] ?? "",
      area: json['area'] ?? "",
      seasonId: json['seasonId'] ?? "",
    );
  }
  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'name': name,
      'seasonId': seasonId,
      'description': description,
      'area': area,
    };
  }
}
